packages/next/src/server/use-cache/cache-life.ts

```diff
@@ -0,0 +1,141 @@
+        'Pass `Infinity` instead of `false` if you want to cache on the client forever ' +
+          'without checking with the server.'
+      )
+    } else if (typeof profile.stale !== 'number') {
+      throw new Error('The stale option must be a number of seconds.')
+    }
+  }
+  if (profile.revalidate !== undefined) {
+    if ((profile.revalidate as any) === false) {
+      throw new Error(
+        'Pass `Infinity` instead of `false` if you do not want to revalidate by time.'
+      )
+    } else if (typeof profile.revalidate !== 'number') {
+      throw new Error('The revalidate option must be a number of seconds.')
+    }
+  }
+  if (profile.expire !== undefined) {
+    if ((profile.expire as any) === false) {
+      throw new Error(
+        'Pass `Infinity` instead of `false` if you want to cache on the client forever ' +
+          'without checking with the server.'
+      )
+    } else if (typeof profile.expire !== 'number') {
+      throw new Error('The expire option must be a number of seconds.')
+    }
+  }
+
+  if (profile.revalidate !== undefined && profile.expire !== undefined) {
+    if (profile.revalidate > profile.expire) {
+      throw new Error(
+        'If providing both the revalidate and expire options, ' +
+          'the expire option must be greater than the revalidate option.' +
+          'The expire option indicates how many seconds from the start ' +
+          'until it can no longer be used.'
+      )
+    }
+  }
+
+  if (profile.stale !== undefined && profile.expire !== undefined) {
+    if (profile.stale > profile.expire) {
+      throw new Error(
+        'If providing both the stale and expire options, ' +
+          'the expire option must be greater than the stale option.' +
+          'The expire option indicates how many seconds from the start ' +
+          'until it can no longer be used.'
+      )
+    }
+  }
+}
+
+export function cacheLife(profile: CacheLifeProfiles | CacheLife): void {
+  if (!process.env.__NEXT_DYNAMIC_IO) {
+    throw new Error(
+      'cacheTag() is only available with the experimental.dynamicIO config.'
+    )
+  }
+
+  const workUnitStore = workUnitAsyncStorage.getStore()
+  if (!workUnitStore || workUnitStore.type !== 'cache') {
+    throw new Error(
+      'cacheLife() can only be called inside a "use cache" function.'
+    )
+  }
+
+  if (typeof profile === 'string') {
+    const configuredProfile = cacheLifeProfileMap.get(profile)
+    if (configuredProfile === undefined) {
+      if (cacheLifeProfileMap.has(profile.trim())) {
+        throw new Error(
+          `Unknown cacheLife profile "${profile}" is not configured in next.config.js\n` +
+            `Did you mean "${profile}" without the spaces?`
+        )
+      }
+      throw new Error(
+        `Unknown cacheLife profile "${profile}" is not configured in next.config.js\n` +
+          'module.exports = {\n' +
+          '  cacheLife: {' +
+          '    "...": ...' +
+          '  }' +
+          '}'
```

Comment:
for the future we should expand the ts plugin to generate this type so we can fail if you pass an invalid string statically